frontend/pulse3d/src/components/table/ColumnHead.js

```diff
@@ -23,41 +23,82 @@ export default function ColumnHead({
   setRightNeighbor,
   rightWidth,
   last,
+  filterColumn,
+  setSortColumns,
+  sortColumn,
 }) {
   const [input, setInput] = useState("");
   const [initialX, setInitialX] = useState();
+  const [order, setOrder] = useState("");
+  useEffect(() => {
+    if (sortColumn !== columnName) {
+      setOrder("");
+    }
+  }, [sortColumn]);
+  useEffect(() => {
+    if (filterColumn !== columnName) {
+      setInput("");
+    }
+  }, [filterColumn]);
   useEffect(() => {
     setFilterString(input);
-    setFilterColumn(columnName);
   }, [input]);
 
   const updateWidth = (e) => {
-    const difference = e.clientX - initialX;
+    const difference = parseInt((e.clientX - initialX) / 10);
     const newWidth = parseInt(width) + difference;
     const newNeighborWidth = parseInt(rightWidth) - difference;
     //prevent user from making columns to small
-    if (newWidth < 150 || newNeighborWidth < 150) {
+    // if (newWidth < 150 || newNeighborWidth < 150) {
+    //   return;
+    // }
+    //steps
+    if (Math.abs(difference) < 3 || newWidth < 9 || newNeighborWidth < 9) {
       return;
     }
-    setSelfWidth(`${newWidth}px`);
-    setRightNeighbor(`${newNeighborWidth}px`);
+    setSelfWidth(`${newWidth}%`);
+    setRightNeighbor(`${newNeighborWidth}%`);
     setInitialX(e.clientX);
   };
   return (
     <Container>
       <FilterInput
+        value={input}
         type="text"
         placeholder={title}
-        onClick={(e) => {
-          e.stopPropagation();
-        }}
         onChange={(e) => {
           setInput(e.target.value);
+          setFilterColumn(columnName);
+        }}
+        onClick={(e) => {
+          e.stopPropagation();
         }}
       />
-
-      <SortByAlphaIcon />
-      {
+      {order === "" ? (
+        <EjectIcon
+          onClick={() => {
+            setOrder("asc");
+            setSortColumns(columnName);
+          }}
+          style={{ color: "gray" }}
+        />
+      ) : null}
+      {order === "asc" && (
```

Comment:
Does this return the `EjectIcon` with an && conditional? I'm more curious, I didn't know you could do that

Reply:
Yes. This is JSX specific I think.  https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

Reply:
Yeah that's cool, I'll start doing that